app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt

```diff
@@ -0,0 +1,121 @@
+package com.jetbrains.handson.mpp.mobile
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.content.Intent
+import android.text.format.DateFormat
+import android.widget.DatePicker
+import android.widget.TextView
+import android.widget.TimePicker
+import kotlinx.android.synthetic.main.activity_pop_up_window.*
+import java.util.*
+
+class AdvancedSearchActivity : AppCompatActivity(), DatePickerDialog.OnDateSetListener,
+        TimePickerDialog.OnTimeSetListener {
+        lateinit var textView: TextView
+        lateinit var button: Button
+        var day = 0
+        var month: Int = 0
+        var year: Int = 0
+        var hour: Int = 0
+        var minute: Int = 0
+        var myDay = 0
+        var myMonth: Int = 0
+        var myYear: Int = 0
+        var myHour: Int = 0
+        var myMinute: Int = 0
+        var timeChanged = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
This function is quite long, and a bit confusing - can you extract any of it into functions that handle say setting all the listeners, or setting up specific parts of the screen?